### DISCUSSION TITLE
Can HTTP-Push Connection Be Established with an Unsecured API (No Certificates/Keys)?

### DISCUSSION THREAD

**Discussion by BHUVANESHWARI967** [eclipse-ditto/ditto#2031] [General]
Hello everyone,

I am trying to establish an HTTP-push connection to an API that does not require any form of certificate-based authentication (i.e., no client certificates or keys). The API is unsecured and only requires basic setup without OAuth, TLS certificates, or HMAC.

Is it possible to create a Ditto HTTP-push connection without providing certificates or keys? If so, are there any specific configuration steps I should follow?

Here’s an example of what I want to achieve:

No SSL certificates or client authentication.
Only need to push data to a basic HTTP endpoint.

example connection Json tried:

{

        "id": "http-api-connection-001",
        "name": "http-push-test",
        "connectionType": "http-push",
        "connectionStatus": "open",
        "failoverEnabled": true,
        "uri": "http://localhost:5000",
        "specificConfig": {
            "parallelism": "1"
        },
        "sources": [],
        "targets": [
            {
                "address": "POST:/logdata",
                "topics": [
                    "_/_/things/twin/events"
                ],
                "authorizationContext": ["nginx:ditto"]
            }
        ]
    }
    
    the api creation is,
    
`from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route('/logdata', methods=['POST'])
def log_data():
    data = request.json
    print(f"Received data: {data}")  # Logs the data received from the request
    return jsonify({"status": "success", "message": "Data logged successfully"}), 200

if __name__ == '__main__':
    app.run(debug=True, port=443)
`

I would appreciate any guidance or examples on how to configure this kind of HTTP-push connection.

Thank you in advance!

**Comment by thjaeckle** [comment]
Hi.

> I am trying to establish an HTTP-push connection to an API that does not require any form of certificate-based authentication (i.e., no client certificates or keys). The API is unsecured and only requires basic setup without OAuth, TLS certificates, or HMAC.

That is no problem ..

I however highly doubt that `localhost` will work in your case (because that resolves to the host where the Ditto `connectivity` service is running).
Do you run the HTTP endpoint in a docker network? Then you should be able to use its service name to address it.

**Comment by BHUVANESHWARI967** [comment]
Hi Thomas, 

Thank you for answering. Apologies for multiple posting. 

I have tried creating a docker image of this script and deployed in an ec2 server where ditto exists. Then also the connection was misconfigured for me.

**Reply by thjaeckle** [reply]
Still, `localhost` will only work when this script would run in the **same** docker container as the Ditto connectivity service, which does not make sense.

**Reply by BHUVANESHWARI967** [reply]
Oh, my bad I missed some details. I am running docker container in the port 31021 which is opened and I have gave <privateIp>:31021 in the uri.

**Reply by BHUVANESHWARI967** [reply]
> Oh, my bad I missed some details. I am running docker container in the port 31021 which is opened and I have gave :31021 in the uri.

I have deployed in kubernetes and the connection creation was successful. Thank you. Will try with other connections!
